sources/academy/tutorials/api/retry_failed_requests.md

```diff
@@ -0,0 +1,61 @@
+---
+title: How to retry failed requests
+description: Learn how to resurrect your run but retrying only failed requests
+sidebar_position: 6
+slug: /api/retry-failed-requests
+---
+
+**Learn how to re-scrape only failed requests in your run.**
+
+---
+
+There are many reasons why requests for a scraper could fail. The most popular causes are different page layouts or proxy blocking issues ([check here on how to effectively analyze errors](https://docs.apify.com/academy/node-js/analyzing-pages-and-fixing-errors)). Both Apify and [Crawlee](https://crawlee.dev/) allow you to restart your scraper run from the point where it ended but there is no native functionality to re-scrape only failed requests. Usually, you also want to first analyze the problem, update the code and build it before trying again.
+
+If you attempt to re-run an already finished run, it will likely immediately finish because all the requests in the [request queue](https://crawlee.dev/docs/3.7/guides/request-storage) are marked as handled. So you need to update the failed requests in the queue to be marked as pending again.
+
+The additional complication is that the [Request](https://crawlee.dev/api/core/class/Request) object doesn't have anything like `isFailed` property, we have to approximate it using other fields. Fortunately, we can use the `errorMessages` and `retryCount` properties to identify failed requests. Unless the user explicitly overrode these properties, we can identify failed requests by having a larger amount of `errorMessages` than `retryCount`. That happens because the last error that doesn't cause a retry anymore is added to `errorMessages`.
```

Comment:
```suggestion
The additional complication is that the [Request](https://crawlee.dev/api/core/class/Request) object doesn't have anything like the `isFailed` property. We have to approximate it using other fields. Fortunately, we can use the `errorMessages` and `retryCount` properties to identify failed requests. Unless the user explicitly has overridden these properties, we can identify failed requests with a larger amount of `errorMessages` than `retryCount`. That happens because the last error that doesn't cause a retry anymore is added to `errorMessages`.
```